proto-lens/src/Data/ProtoLens/Encoding/Parser.hs

```diff
@@ -68,6 +71,12 @@ getBytes :: Int -> Parser ByteString
 getBytes n = withSized n "getBytes: Unexpected end of input"
                     $ \pos -> packCStringLen (castPtr pos, n)
 
+getText :: Int -> Parser Text
+getText n = do
+  r <- withSized n "getText: Unexpected end of input" $ \pos ->
+          decodeUtf8' <$> B.unsafePackCStringLen (castPtr pos, n)
```

Comment:
The `unsafe` here peaked my curiosity so I went looking and convinced myself that `withSized` does enough check to ensure safety of touching [pos, pos+n) bytes.